numpy/core/arrayprint.py

```diff
@@ -438,14 +438,15 @@ def _array2string(a, options, separator=' ', prefix=""):
     # The formatter __init__s in _get_format_function cannot deal with
     # subclasses yet, and we also need to avoid recursion issues in
     # _formatArray with subclasses which return 0d arrays in place of scalars
-    a = asarray(a)
+    data = asarray(a)
+    if a.shape == ():
```

Comment:
What if `a` doesn't have `shape`? Should this be a try?

Reply:
I assume this is checked somewhere up above?

Reply:
Looks like `arraystring` requires an `ndarray`, or subclass thereof, as `a`, but that is not checked for. The documentation says `array_like`, which looks incorrect.